tests/lib/eslint/eslint.js

```diff
@@ -6181,6 +6222,50 @@ describe("ESLint", () => {
 
                 });
 
+                it("should load a TS config file when --experimental-strip-types is enabled", async () => {
+
+                    const originalProcessEnv = process.env;
+
+                    process.env = { ...process.env, NODE_OPTIONS: "--experimental-transform-types" };
```

Comment:
Overwriting `process.env` has no effect. We need to launch a new Node.js process and pass the `--experimental-transform-types` flag.